server/api/shared.py

```diff
@@ -553,18 +560,25 @@ def compute_sample_batches(
     }
 
     previously_sampled_batch_keys: List[sampler.BatchKey] = list(
-        SampledBatchDraw.query.join(Batch)
-        .join(Jurisdiction)
-        .filter_by(election_id=election.id)
+        Batch.query.join(Jurisdiction)
+        .filter(Jurisdiction.election_id == contest.election_id)
+        .join(SampledBatchDraw)
+        # Don't include non-RLA batches
+        .filter(
+            and_(
+                SampledBatchDraw.contest_id == contest.id,
+                SampledBatchDraw.ticket_number != EXTRA_TICKET_NUMBER,
```

Comment:
We were previously forgetting to filter out "extra" non-RLA batches here, but it didn't matter because the only audit that we'd ever run with extra batches was configured to automatically end after one round. And this retrieval is only relevant when sampling batches for a second round.